worker/worker_test.go

```diff
@@ -0,0 +1,124 @@
+package worker
+
+import (
+	"context"
+	"time"
+
+	rhpv2 "go.sia.tech/core/rhp/v2"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/blake2b"
+	"lukechampine.com/frand"
+)
+
+type (
+	test interface {
+		Fatal(...any)
+	}
+
+	testWorker struct {
+		t test
+		*worker
+
+		cs *contractStoreMock
+		os *objectStoreMock
+		hs *hostStoreMock
+
+		dlmm *memoryManagerMock
+		ulmm *memoryManagerMock
+
+		hm *testHostManager
+	}
+)
+
+func newTestWorker(t test) *testWorker {
+	// create bus dependencies
+	cs := newContractStoreMock()
+	os := newObjectStoreMock(testBucket)
+	hs := newHostStoreMock()
+
+	// create worker dependencies
+	b := newBusMock(cs, hs, os)
+	dlmm := newMemoryManagerMock()
+	ulmm := newMemoryManagerMock()
+
+	// create worker
+	w, err := New(blake2b.Sum256([]byte("testwork")), "test", b, time.Second, time.Second, time.Second, time.Second, 0, 0, 1, 1, false, zap.NewNop())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// override managers
+	hm := newTestHostManager(t)
+	w.priceTables.hm = hm
+	w.downloadManager.hm = hm
+	w.downloadManager.mm = dlmm
+	w.uploadManager.hm = hm
+	w.uploadManager.mm = ulmm
+
+	return &testWorker{
+		t,
+		w,
+		cs,
+		os,
+		hs,
+		dlmm,
+		ulmm,
+		hm,
+	}
+}
+
+func (w *testWorker) addHosts(n int) (added []*testHost) {
```

Comment:
I'd expose methods that are meant to be called in tests and not expose the ones that are only meant to be used internally within the mock.